### PR TITLE
support translate other than <p>

### PR DESCRIPTION
**Pull request by hleft**
compare

如图对比， 补充了翻译

### CODE REVIEW

**Comment by yihong0618** [comment]
cool  but FYI I did not add this because blockquote  always are code....
They do not need to translate
@typebrook @hleft  I think we can add this to an option in args?
like we discuss on tw `div` tag

**Comment by hleft** [comment]
@yihong0618 Yes, it is better to add option support to configure the tags to be translated in addition to the p tag

**Comment by yihong0618** [comment]
agree～

**Comment by hleft** [comment]
@yihong0618  Then I will add a new pr

**Comment by yihong0618** [comment]
@hleft just add more commit on the same PR is OK.

**Comment by typebrook** [comment]
>  I think we can add this to an option in args

Agree. I suggest includes tags like `<h1>`, `<h2>`, `<h3>`, `<div>`, `<td>`, `<blockquote>` and <li>.

Proposal:

- Option `--tag-all` to translate all included tags
- Option `--tag` for custom tags, usage would be like:
```python
python3  make_book.py \
  --book_name test_books/foo.epub \
  --openai_key ${OPPEN_API_KEY} \ 
  --language zh-hant \
  --tag div \
  --tag td 
```

I can make a PR for this tonight, or would you like to make your hands dirty now?

**Comment by hleft** [comment]
@typebrook  Because I'm not sure what all the tags should be, I just added a custom tag . 

```
  --translate-tags TRANSLATE_TAGS
                        example --translate-tags p,blockquote
```

**Comment by yihong0618** [comment]
please fix the lint
@typebrook do you have time help me to review it, since its your idea too~

**Comment by typebrook** [comment]
Work as babysitter right now, will get back to this tonight

File: book_maker/cli.py [hunk 1]
```diff
@@ -119,6 +126,7 @@ def main():
         model_api_base=model_api_base,
         is_test=options.test,
         test_num=options.test_num,
+        translate_tags=options.translate_tags,
```
Comment (yihong0618) [inline book_maker/cli.py:129]: make `p` as default

**Comment by jeffery9** [comment]
great feature.

**Comment by hleft** [comment]
Sorry, I pushed another pr to this pr... wait for me to see how to modify it

**Comment by hleft** [comment]
I accidentally pushed the content of another PR to here, and it has been revised back, please act as if nothing happened..

**Comment by typebrook** [comment]
@hleft 
Now I am testing new options with this epub file, which put most of its contents into `<div>`:
https://topo.tw/public/tmp/The_World.epub

This file comes from another twitter user, read original tweets at:
https://twitter.com/typebrook/status/1633313460737150977

Looks good now!
Screenshot_2023-03-08-20-08-38_1920x1080
Screenshot_2023-03-08-20-08-53_1920x1080


@yihong0618 Do you think it is a good idea to add this epub file into `test_books`

**Comment by yihong0618** [comment]
It is better to Choose Books Without Copyright

**Comment by typebrook** [comment]
@yihong0618 
> It is better to Choose Books Without Copyright

Good point, I'll check if any books in https://www.gutenberg.org use tags other than `<p>`.

**Comment by typebrook** [comment]
How about bible (Latin)? This may be a proper test both for Chinese/English native reader:
https://www.gutenberg.org/ebooks/826
https://zh.wikipedia.org/zh-tw/以斯帖記

EBooks from **gutenberg** puts disclaims in `<div>`, and this book also put story title into `<h5>`. I think it is a good choice for the demo of option `--translate-tags`!

@hleft 
If @yihong0618 is fine with this, please add [this book] into `test_books/` in a new commit.

[this book]: https://www.gutenberg.org/ebooks/826.epub.noimages

**Comment by yihong0618** [comment]
ok

**Comment by hleft** [comment]
@typebrook @yihong0618  in .gitignore `/test_books/*.epub`,  Should I modify gitignore？ how to add this book?

**Comment by typebrook** [comment]
>  Should I modify gitignore？
No

> how to add this book?
```bash
cd test_books/
curl -Lo Liber_Esther.epub https://www.gutenberg.org/ebooks/826.epub.noimages
```

@hleft @yihong0618 
Test case finished, LGTM. I think after adding new ebook into `test_books/`, this PR is ready to merge.
```bash
python3 make_book.py \                                                                                       1
  --book_name /home/pham/Downloads/Liber_Esther.epub \
  --model gpt3 \
  --no_limit \
  --language "Traditional Chinese" \
  --translate-tags div,p,h1,h2,h3,h4,h5
```

Screenshot_2023-03-08-21-31-57_1920x1080
Screenshot_2023-03-08-21-32-06_1920x1080
Screenshot_2023-03-08-21-32-24_1920x1080



@hleft 
Also, could you modify PR title into **"support translate other than \<p\>"** ? This may helps future development.



P.S. I found `<br>`(it means newline) in each tags are not properly processed, but this is another issue.

**Comment by hleft** [comment]
@typebrook  .gitignore contain 

```
/test_books/*.epub
test_books/
```

still ignore

**Comment by typebrook** [comment]
@yihong0618 I think it is an accident in ab3bb4b, we should remove this line in `.gitignore`

**Comment by typebrook** [comment]
Oh...you intentionally add this line in e9066d0

@hleft Great job! PR is LGTM now.

**Comment by yihong0618** [comment]
Seems that we need add another test

**Comment by typebrook** [comment]
Wait! Now we just pick book of test by filename? (`head -n 1`)
https://github.com/yihong0618/bilingual_book_maker/blob/b25c4ca873fb1ef357884663418a3d2d16b09a02/.github/workflows/make_test_ebook.yaml#L21

**Comment by yihong0618** [comment]
This is not my code....
I think head -n 1 is fine...

**Comment by yihong0618** [comment]
And we need mention it in README

**Pull request merged**